### ISSUE TITLE
Hidden (Offscreen Text) Reported as 1:NaN Contrast Error in AAA Mode

### ISSUE THREAD

**Issue by tmcconechy** [squizlabs/HTML_CodeSniffer#119]
1. Open this jsbin http://jsbin.com/buwefuduga/1/
2. Run Code Sniffer Bookmarklet
3. Set mode to WCAG2AAA (Issue is on Triple A only)
4. Note the errors... In particular:

This element has insufficient contrast at this conformance level. Expected a contrast ratio of at least 7:1, but text in this element has a contrast ratio of NaN:1.

This is problematic because a) its reporting NaN so something must be off.. b) This one should pass since its white on black.

It appears like the issue is related to absolutely positioned elements but thats a guess.

**Comment by lwright-sq** [comment]
Hi there,

I suspect that it's not handling background colour being set as a word "black" rather than a hex colour vlaue. If you set it as "#000000" do you get the same error?

If the problem is indeed with "black", I'll have to take a look at what can be done with this.

**Comment by tmcconechy** [comment]
Hi, No its not black i updated the bin.. http://jsbin.com/xigixoyeka/1/ I think its the absolutely positioned element its having issues with..

The issue seems pretty prevalent with the AAA checker. For example i see a bunch of element has a contrast ratio of NaN:1 on the log for sites like:

http://www.cnn.com/ or even
http://www.squiz.net/us/platform/matrix (although this site has error on a view statically position elements)

**Comment by telesma** [comment]
Works for me with "position: fixed" instead of "position: absolute"

**Comment by tmcconechy** [comment]
Yes, definitely reproducible on position: absolute, actually really effects usefulness for AAA level. Do not really understand how this code is put together or i would try a pull request

**Comment by lwright-sq** [comment]
Merged in @telesma 's pull request at #135. This was a check I'd put into the AA test (1.4.3) but not into the AAA test (1.4.6) for some reason. Whoops. my bad. Thanks for the pull req.

**Comment by tmcconechy** [comment]
Thanks. What would i have to do to take this in the bookmarklet in my browser?

**Comment by dak** [comment]
Is this issue fixed in the current version?  I'm still seeing the exact same issue with an offscreen element in AAA mode only.

I'm also seeing a related issue in AA and AAA mode with an absolutely positioned element (removing the absolute positioning or adding a background-color to the element fixes it).  In this case, the wrong contrast is being reported, rather than showing NaN.

I can try to create test case for each issue if they are unknown issues.

**Comment by lwright-sq** [comment]
Hi Derek (@dak),

Were you using a version compiled yourself, or still using the bookmarklet?
The bookmarklet didn't get updated until yesterday (19 Jan), and should have this commit - can you please re-test?

Test cases would also help if you can provide those. Thanks.

**Comment by tmcconechy** [comment]
Great, seems working on the bookmarklet now :)

**Comment by dak** [comment]
@lwright-sq I seem to still be seeing this issue in AA and AAA mode for absolutely positioned elements (see screenshots below — adding `background-color: white` fixes the error), although the error with offscreen text does appear to be fixed.  Should I open a new issue?

There also appears to be a new issue with the updated code where google analytics code is being injected with an <img> that lacks alt text every time the bookmarklet is opened, causing more and more errors to be reported each time.

![image](https://cloud.githubusercontent.com/assets/1371232/12469482/1e0fc9ac-bfba-11e5-9b45-6a218a48f3ea.png)
![image](https://cloud.githubusercontent.com/assets/1371232/12469490/2e3434c6-bfba-11e5-936d-799e15f75d29.png)

**Comment by tmcconechy** [comment]
Yeah i noticed the second part and raised it here: https://github.com/squizlabs/HTML_CodeSniffer/issues/148
